A19-tls-credential-API

```diff
@@ -0,0 +1,281 @@
+   void *cb_user_data;
+   bool *result;
+   const char *target_name;
+   const char *certificate;
+   grpc_status_code *status;
+   const char **error_details;
+} grpc_tls_server_authorization_check_arg; 
+
+/** Create a grpc_tls_server_authorization_check_config instance.
+    - config_user_data is config-specific, read-only user data
+      that works for all channels created with a credential using the config.
+    - schedule is a pointer to an application-provided callback used to invoke server
+      authorization check API. The implementation of this method has to be non-blocking,
+      but can be performed synchronously or asynchronously.
+      1) If processing occurs synchronously, it populates arg->result,
+      arg->status, and arg->error_details and returns zero.
+      2) If processing occurs asynchronously, it returns a non-zero value.
+      The application then invokes arg->cb when processing is completed. Note that
+      arg->cb cannot be invoked before schedule API returns.
+    - cancel is a pointer to an application-provided callback used to cancel a server
+      authorization check request scheduled via an asynchronous schedule API.
+      arg is used to pinpoint an exact check request to be cancelled.
+      The operation may not have any effect if the request has already been processed.
+    - destruct is a pointer to an application-provided callback used to clean up any data
+      associated with the config.
+*/
+GRPCAPI grpc_tls_server_authorization_check_config*
+                                      grpc_tls_server_authorization_check_config_create(
+                                      const void* config_user_data, 
+                                      int (*schedule)(const void* config_user_data,
+                                           grpc_tls_server_authorization_check_arg* arg), 
+                                      void (*cancel)(const void* config_user_data,
+                                            grpc_tls_server_authorization_check_arg* arg),
+                                      void (*destruct)(const void* config_user_data));
+
+/** Destroy a grpc_tls_server_authorization_check_config instance. */
+GRPCAPI void grpc_tls_server_authorization_check_config_destroy(grpc_tls_server_authorization_check_config* config);
+```
+
+## Rationale
+
+Regarding grpc_tls_ctx_customize_config, the config is used when a caller wants to
+configure an underlying TLS context with customized primitives. For example, there could
+be a use case in which raw private keys are not directly accessible (e.g., hardware-backed) to a caller,
+but private key methods (e.g., “Sign” functions) are, in which case the config provides a means
+to customize an TLS context with an SSL_PRIVATE_KEY_METHOD object and associated signing algorithms.
+
+## Implementation
+
+The implementation of this proposal is straightforward as all API changes will be
+made to include/grpc/grpc_security.h.
```

Comment:
Also mention that when this proposal is implemented, we will start deprecating old SSL credentials API. User of gRPC TLS transport security shall start using this new API.